src/extension.ts

```diff
@@ -99,127 +85,37 @@ export class WatermelonTreeDataProvider
-      }
-      let uniqueBlames = await getBlame(gitAPI, startLine, endLine);
-      let issuesWithTitlesAndGroupedComments = await getPRsToPaintPerSHAs({
-        arrayOfSHAs,
-        email: session?.account.label || "",
-        owner,
-        repo,
-        repoSource,
-      });
-      let sortedPRs: any[] = [];
-      if (Array.isArray(issuesWithTitlesAndGroupedComments)) {
-        sortedPRs = issuesWithTitlesAndGroupedComments?.sort(
-          (a: any, b: any) => b?.comments?.length - a?.comments?.length
-        );
-      }
-
-      const parsedCommitObject = new Object(uniqueBlames[0]) as {
-        date: string;
-        message: string;
-        author: string;
-        email: string;
-        commit: string;
-        body: string;
-        sha: string;
-      };
-      const parsedMessage = parsedCommitObject.message;
+    let uniqueBlames;
 
-      debugLogger(`parsedMessage: ${parsedMessage}`);
-      if (!session) {
-        setLoggedIn(false);
-        return items;
-      }
-      let itemPromises = [
-        getHubLabBucketItems(issuesWithTitlesAndGroupedComments, repoSource),
-        getGitItems(uniqueBlames),
-        getJiraItems(
-          sortedPRs[0]?.title || parsedMessage,
-          session.account.label
-        ),
-        getSlackItems(
-          sortedPRs[0]?.title || parsedMessage,
-          session.account.label
-        ),
-        getCodeContextSummary(
-          sortedPRs[0]?.title || parsedMessage,
-          sortedPRs[0]?.body || parsedCommitObject.body,
-          session.account.label
-        ),
-      ];
-      let results = await Promise.all(itemPromises);
-      results.forEach((result) => {
-        items.push(...result);
-      });
-      reporter?.sendTelemetryEvent("getCodeContext");
-      return items;
-    } else {
-      vscode.commands.executeCommand("watermelon.multiSelect");
-      arrayOfSHAs = await getSHAArray(
-        (startLine && startLine > 1 ? startLine - 1 : startLine) ?? 1,
-        endLine
-          ? endLine + 1
-          : vscode.window.activeTextEditor?.document.lineCount ?? 2,
-        vscode.window.activeTextEditor?.document.uri.fsPath,
-        gitAPI
+    if (
+      (startLine === undefined && endLine === undefined) ||
+      startLine === endLine
+    ) {
+      uniqueBlames = await getBlame(
+        gitAPI,
+        1,
+        vscode.window.activeTextEditor?.document.lineCount ?? 2
       );
-      if (!arrayOfSHAs?.length) {
-        arrayOfSHAs = await getSHAArray(
-          1,
-          vscode.window.activeTextEditor?.document.lineCount ?? 2,
-          vscode.window.activeTextEditor?.document.uri.fsPath,
-          gitAPI
-        );
-      }
+    } else {
+      uniqueBlames = await getBlame(gitAPI, startLine, endLine);
+    }
 
-      let issuesWithTitlesAndGroupedComments = await getPRsToPaintPerSHAs({
-        arrayOfSHAs,
+    if (!session) {
+      setLoggedIn(false);
+      return items;
+    }
+    const [contextItems] = await Promise.all([
```

Comment:
Nice! This also looks syntactically way more organized. Question: We depend on a lot of calls to different APIs and we want to show the ones that succeed even if one fails (eg: Slack is down, Open AI is down). Would promise.allSettled( ) suffice this better?

Just a minor detail that regardless of the answer, gives more context.

Reply:
For now, we show only the ones that return properly. We could make the UI show the errors that the backend sends.